core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -272,18 +272,25 @@ object AdminUtils extends Logging with AdminUtilities {
                     numPartitions: Int = 1,
                     replicaAssignmentStr: String = "",
                     checkBrokerAvailable: Boolean = true,
-                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced) {
+                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced,
+                    validateOnly: Boolean = false) {
+    if (zkUtils.pathExists(ZkUtils.ReassignPartitionsPath)) {
+      // We prevent addition partitions while a reassignment is in progress, since
+      // during reassignment there is no meaningful notion of replication factor
+      throw new ReassignmentInProgressException("A partition reassignment is in progress.")
+    }
+
     val existingPartitionsReplicaList = zkUtils.getReplicaAssignmentForTopics(List(topic))
     if (existingPartitionsReplicaList.isEmpty)
-      throw new AdminOperationException("The topic %s does not exist".format(topic))
+      throw new InvalidTopicException(s"The topic $topic does not exist")
 
     val existingReplicaListForPartitionZero = existingPartitionsReplicaList.find(p => p._1.partition == 0) match {
-      case None => throw new AdminOperationException("the topic does not have partition with id 0, it should never happen")
+      case None => throw new AdminOperationException("The topic does not have partition with id 0, it should never happen.")
```

Comment:
Maybe we can rewrite this along the lines of s"Unexpected replica assignment for topic '$topic', partition id 0 is missing. Assignment: $existingPartitionsReplicaList".